src/execution/base_aggregate_hashtable.cpp

```diff
@@ -20,11 +20,15 @@ BaseAggregateHashTable::BaseAggregateHashTable(BufferManager &buffer_manager, ve
                                                vector<LogicalType> payload_types_p,
                                                vector<AggregateObject> aggregate_objects)
     : buffer_manager(buffer_manager), aggregates(move(aggregate_objects)), group_types(move(group_types_p)),
-      payload_types(move(payload_types_p)), group_width(0), group_padding(0), payload_width(0) {
+      payload_types(move(payload_types_p)), group_width(0), group_mask_width(0), group_padding(0), payload_width(0) {
 
+	// Null mask at the front - 1 bit per value.
+	group_mask_width = ValidityData::EntryCount(group_types.size()) * sizeof(validity_t);
```

Comment:
Not for this PR, but it would be nice if the payload null masks could also be included here so the individual aggregates don't need to keep track of this anymore.

Reply:
Aggregate states don't have `NULL` flags - it is only when you call `Finalize`that a `NULL` might be generated. AmI missing something?

Reply:
Aggregates need to keep track of whether or not any non-NULL value was input, for example `sum(x)` needs to know if any non-null value of `x` was passed in. Currently the aggregates keep track of this themselves (using the `isset` flag in the sum, for example, or a separate `count`), but we could envision keeping track of this in the hash table itself.

Reply:
We could, but that tangles aggregate functions into hash tables, which are not the only clients. I'd rather the hash table knew less about how aggregation works outside of the API. I had to get rid of one bit of code that assumed that the `Finalize` method for `COUNT` was a NOP...